store/conf/attrs/zimbra-attrs.xml

```diff
@@ -9802,4 +9802,37 @@ TODO: delete them permanently from here
     <defaultCOSValue>TRUE</defaultCOSValue>
     <desc>Whether Powerpaste preference is enabled by user or admin</desc>
 </attr>
+
+<attr id="3086" name="zimbraFeatureChangeUseImapEnabled" type="boolean" cardinality="single" optionalIn="account,cos" flags="accountInfo,accountInherited,domainAdminModifiable" since="8.8.15">
+  <defaultCOSValue>FALSE</defaultCOSValue>
+  <desc>
+    whether to allow end user to disable IMAP feature.
+    If it is set to TRUE, IMAP feature is available only when both zimbraImapEnabled and zimbraPrefUseImap are TRUE.
+    If it is set to FALSE, zimbraPrefUseImap is ignored.
+  </desc>
+</attr>
+
+<attr id="3087" name="zimbraFeatureChangeUsePop3Enabled" type="boolean" cardinality="single" optionalIn="account,cos" flags="accountInfo,accountInherited,domainAdminModifiable" since="8.8.15">
+  <defaultCOSValue>FALSE</defaultCOSValue>
+  <desc>
+    whether to allow end user to disable POP3 feature.
+    If it is set to TRUE, POP3 feature is available only when both zimbraPop3Enabled and zimbraPrefUsePop3 are TRUE.
+    If it is set to FALSE, zimbraPrefUsePop3 is ignored.
+  </desc>
+</attr>
+
+<attr id="3088" name="zimbraPrefUseImap" type="boolean" cardinality="single" optionalIn="account,cos" flags="accountInherited,domainAdminModifiable" since="8.8.15">
```

Comment:
Why are we using 2 attributes? zimbraPrefUseImap and zimbraFeatureChangeUseImapEnabled?

Reply:
In the current design, a setting (checkbox) of zimbraPrefUseImap is shown in Preferences when both zimbraImapEnabled and zimbraFeatureChangeUseImapEnabled are TRUE.

* zimbraImapEnabled: configured by an administator
* zimbraFeatureChangeUseImapEnabled: whether to allow a user to disable imap
* zimbraPrefUseImap: configured by an account

IMAP is enabled/disabled as follows:

|zimbraImapEnabled|zimbraFeatureChangeUseImapEnabled|zimbraPrefUseImap|IMAP access|
--------------------|----------------------------------------|--------------------|------------|
|TRUE|TRUE|TRUE|Allowed|
|TRUE|TRUE|FALSE|Denied|
|TRUE|FALSE|(any)|Allowed|
|FALSE|(any)|(any)|Denied|

Reply:
**zimbraFeatureChangeUseImapEnabled: whether to allow a user to disable imap**   why do we need this?  
It looks to complicated that admin need to decide if user can disable/enable imap or pop3.  


zimbraImapEnabled   (Only for the Admin | zimbraPrefImapEnabled   (User/COS) | IMAP access
-- | -- | --
TRUE | TRUE | TRUE
TRUE | FALSE | FALSE
FALSE | TRUE | FALSE
FALSE | FALSE | FALSE

Reply:
I thought there might be a client who wouldn't like a user to disable imap/pop3. If it is good to show the settings in Preferences always, we can remove zimbraFeatureChangeUseImapEnabled.

Reply:
I do not think we should make if so complex that admin should decide if preferences should be available to end user or not.  Mostly zimbraPrefImapEnabled  will be unset for the user.  
As long as setting   zimbraPrefImapEnabled impact will be at user level we should go with simple approach. 
zimbraImapEnabled (Only for the Admin). and zimbraPrefImapEnabled (User/COS)

Reply:
I agree.

Reply:
Fixed.